File: check_db.js
/* eslint-disable @typescript-eslint/no-require-imports */
require('dotenv').config({ path: '.env.local' })
const { createClient } = require('@supabase/supabase-js')

const url = process.env.NEXT_PUBLIC_SUPABASE_URL
const key = process.env.SUPABASE_SERVICE_ROLE_KEY

const supabase = createClient(url, key)

async function check() {
    console.log("Checking User: b007d7d0-1735-492e-bd65-396d03ecde7d")
    const { data: user, error: userError } = await supabase.auth.admin.getUserById('b007d7d0-1735-492e-bd65-396d03ecde7d')
    console.log("Auth User Data:", JSON.stringify(user, null, 2))
    if (userError) console.error("Auth User Error:", userError)

    const { data: dbUser, error: dbError } = await supabase.from('users').select('*').eq('id', 'b007d7d0-1735-492e-bd65-396d03ecde7d')
    console.log("Public Users Table Data:", JSON.stringify(dbUser, null, 2))
    if (dbError) console.error("DB Error:", dbError)
}

check()

File: check_db.py
import os
from supabase import create_client, Client

url: str = os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
key: str = os.environ.get("NEXT_PUBLIC_SUPABASE_ANON_KEY")

supabase: Client = create_client(url, key)

print("--- users in public.users ---")
try:
    response = supabase.table("users").select("*").execute()
    print(response.data)
except Exception as e:
    print(f"Error fetching public.users: {e}")

print("--- trying to sign in ---")
try:
    response = supabase.auth.signInWithPassword({
        "email": "mikami@asahipac.co.jp",
        "password": "REPLACE_ME_LATER" # I don't have the password, but I can check if the user exists
    })
    print(response.user)
except Exception as e:
    print(f"Error signing in: {e}")

